src/kontain.c

```diff
@@ -205,10 +205,17 @@ int
 add_kontain_config(libcrun_container_t *container)
 {
   int ret;
+  struct stat statb;
   const char *use_virt = find_annotation(container, APP_KONTAIN_USEVIRT);
   if (use_virt == NULL) {
-    libcrun_warning("Couldn't find %s annotation, using kvm", APP_KONTAIN_USEVIRT);
-    use_virt = "kvm";
+    if (stat("/dev/kvm", &statb) == 0) {
+      use_virt = "kvm";
+    } else if (stat("/dev/kkm", &statb) == 0) {
+      use_virt = "kkm";
+    } else {
+      libcrun_warning("Couldn't detect kvm or kkm, using kvm", APP_KONTAIN_USEVIRT);
+      use_virt = "kvm";
+    }
```

Comment:
I have two issues here.

- We have similar (same?) code in km itself. Feels a bit strange to have it in both places. Why not delegate that decision to km? Plus, if both are available, and payload is a snapshot, km will be able to pick the right one, an krun most likely wont.
- (Assuming it stays here) If both are AVOL, we know km will fail. Why even try?

Reply:
We use this to decide which device file to bind mount into the container.
So, we need to know what to do at container create time.

What is AVOL?

Reply:
I guess if both are available we could bind mount both of them.
I'll do that.

Reply:
AVOL was meant to be AWOL :-)

Also, I guess if there is only one, we should pass that flag to km so that it won't bother looking.

Good point on both present.